lib/acts_as_list/active_record/acts/database_mutexes.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'with_advisory_lock'
+
+module ActiveRecord
+  module Acts #:nodoc:
+    module List #:nodoc:
+      module DatabaseMutexes #:nodoc:
+        def self.included(base)
+          base.prepend(InstanceMethods)
+        end
+
+        module InstanceMethods
+          ActiveRecord::Acts::List::InstanceMethods.public_instance_methods.each do |m|
```

Comment:
Do you think we should be more surgical? Do all the methods require a lock? This is kind of elegant though.

Reply:
Not sure. I guess it does not hurt? we can always be more surgical later on as an optimisation.

Reply:
Looking at the list of methods, there are quite a few read-only methods there. Also, what happens when the methods call other public methods? Just a bit worried about over-locking when it's not necessary. @swanandp, did you have any comments on this?

Reply:
Also, I see @swanandp's commend earlier about wanting locking to be something that users have to enable, so let's stick with that strategy at this point. I'd say let's make that available to configure in an `initializer` and then also individually opt-out per `acts_as_list` call?

Reply:
@brendon: alright, will try to cherry pick the methods, make an initializer and default to false

Reply:
@brendon: I cannot find this `initializer` you are talking about... can you point me to the right place?

Reply:
Hi @Silex :) There isn't one currently. It's more that we'll need a way to tell `ActsAsList` to turn on this feature while the Rails app boots. Looking through the initialisers directory on one of my apps, gem's either provide a block for configuration or just allow one to make calls like:

```
Rails.application.config.action_dispatch.cookies_serializer = :json
```

to change settings.